pkg/api/imagereferencemutators/internalversion/pods.go

```diff
@@ -0,0 +1,421 @@
+
+// GetPodSpec returns a mutable pod spec out of the provided object, including a field path
+// to the field in the object, or an error if the object does not contain a pod spec.
+// This only returns internal objects.
+func GetPodSpec(obj runtime.Object) (*kapi.PodSpec, *field.Path, error) {
+	switch r := obj.(type) {
+	case *kapi.Pod:
+		return &r.Spec, field.NewPath("spec"), nil
+	case *kapi.PodTemplate:
+		return &r.Template.Spec, field.NewPath("template", "spec"), nil
+	case *kapi.ReplicationController:
+		if r.Spec.Template != nil {
+			return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+		}
+	case *extensions.DaemonSet:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *extensions.Deployment:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *extensions.ReplicaSet:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *batch.Job:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *batch.CronJob:
+		return &r.Spec.JobTemplate.Spec.Template.Spec, field.NewPath("spec", "jobTemplate", "spec", "template", "spec"), nil
+	case *batch.JobTemplate:
+		return &r.Template.Spec.Template.Spec, field.NewPath("template", "spec", "template", "spec"), nil
+	case *apps.StatefulSet:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *securityapi.PodSecurityPolicySubjectReview:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *securityapi.PodSecurityPolicySelfSubjectReview:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *securityapi.PodSecurityPolicyReview:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *appsapi.DeploymentConfig:
+		if r.Spec.Template != nil {
+			return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+		}
+	}
+	return nil, nil, errNoPodSpec
+}
+
+// GetPodSpecV1 returns a mutable pod spec out of the provided object, including a field path
+// to the field in the object, or an error if the object does not contain a pod spec.
+// This only returns pod specs for v1 compatible objects.
+func GetPodSpecV1(obj runtime.Object) (*kapiv1.PodSpec, *field.Path, error) {
+	switch r := obj.(type) {
+	case *kapiv1.Pod:
+		return &r.Spec, field.NewPath("spec"), nil
+	case *kapiv1.PodTemplate:
+		return &r.Template.Spec, field.NewPath("template", "spec"), nil
+	case *kapiv1.ReplicationController:
+		if r.Spec.Template != nil {
+			return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+		}
+	case *extensionsv1beta1.DaemonSet:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *extensionsv1beta1.Deployment:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *extensionsv1beta1.ReplicaSet:
```

Comment:
Not to mention this list is not sufficient, but I think I'm fixing that in my PR.